internal/httpproxy/store.go

```diff
@@ -0,0 +1,177 @@
+package httpproxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// Mode definition
+type Mode string
+
+const (
+	// ModePassthrough mode
+	ModePassthrough Mode = "passthru"
+)
+
+// ErrorResponse definition
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// Request definition
+type Request struct {
+	Method  string      `json:"method"`
+	Path    string      `json:"path"`
+	Headers http.Header `json:"header"`
+	Query   url.Values  `json:"query"`
+	Body    []byte      `json:"body"`
+}
+
+// WriteResponse definition
+type WriteResponse struct {
+	StatusCode int               `json:"statuscode"`
+	Headers    http.Header       `json:"header"`
+	Body       WriteResponseBody `json:"body"`
+}
+
+// WriteResponseBody definition
+type WriteResponseBody struct {
+	Offset int `json:"offset"`
+}
+
+// Entry definition
+type Entry struct {
+	Offset   int           `json:"offset"`
+	Request  Request       `json:"request"`
+	Response WriteResponse `json:"response"`
+}
+
+// HTTPServerProxyStore definition
+type Store struct {
+	Name string `json:"name"`
+	Mode Mode   `json:"mode"`
+	Data []Entry
+}
+
+// write stores incoming request data
+func (store *Store) write(w http.ResponseWriter, r *http.Request) {
+
+	reqData := Request{
+		Method:  r.Method,
+		Path:    r.URL.RequestURI(),
+		Headers: r.Header,
+		Query:   r.URL.Query(),
+	}
+
+	offset := len(store.Data)
+
+	resData := WriteResponse{
+		StatusCode: http.StatusOK,
+		Headers: http.Header{
+			"Content-Type": []string{"application/json"},
+		},
+		Body: WriteResponseBody{
+			Offset: offset,
+		},
+	}
+
+	if r.Body != nil {
+		bin, err := ioutil.ReadAll(r.Body)
+		if err != nil {
```

Comment:
Why dont you read this straight into the store? Why the extra `bin` variable here?